Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/CompiledLibrary.kt

```diff
@@ -0,0 +1,281 @@
+        return resolveIdentifier(identifier, IncludeDef::class.java)
+    }
+
+    fun resolveIncludeAlias(identifier: VersionedIdentifier?): String? {
+        return when {
+            identifier != null && library?.includes?.def != null -> {
+                val libraryPath = NamespaceManager.getPath(identifier.system, identifier.id)
+                library!!.includes.def.firstOrNull { it.path == libraryPath }?.localIdentifier
+            }
+            else -> null
+        }
+    }
+
+    fun resolveCodeSystemRef(identifier: String): CodeSystemDef? {
+        return resolveIdentifier(identifier, CodeSystemDef::class.java)
+    }
+
+    fun resolveValueSetRef(identifier: String): ValueSetDef? {
+        return resolveIdentifier(identifier, ValueSetDef::class.java)
+    }
+
+    fun resolveCodeRef(identifier: String): CodeDef? {
+        return resolveIdentifier(identifier, CodeDef::class.java)
+    }
+
+    fun resolveConceptRef(identifier: String): ConceptDef? {
+        return resolveIdentifier(identifier, ConceptDef::class.java)
+    }
+
+    fun resolveParameterRef(identifier: String): ParameterDef? {
+        return resolveIdentifier(identifier, ParameterDef::class.java)
+    }
+
+    fun resolveExpressionRef(identifier: String): ExpressionDef? {
+        return resolveIdentifier(identifier, ExpressionDef::class.java)
+    }
+
+    private fun <T : Element> resolveIdentifier(identifier: String, clazz: Class<T>): T? {
+        return resolve(identifier).resolveIdentifier(clazz)
+    }
+
+    fun resolveFunctionRef(identifier: String): Iterable<FunctionDef> {
+        val results = ArrayList<FunctionDef>()
+        for (ed in library!!.statements.def) {
+            if (ed is FunctionDef && ed.getName() == identifier) {
+                results.add(ed)
+            }
+        }
+
+        return results
+    }
+
+    fun resolveFunctionRef(
+        functionName: String,
+        signature: List<DataType>?
+    ): Iterable<FunctionDef?> {
+        return when (signature) {
+            null -> resolveFunctionRef(functionName)
+            else -> {
+                val cc =
+                    CallContext(
+                        this.identifier!!.id,
+                        functionName,
+                        allowPromotionAndDemotion = false,
+                        allowFluent = false,
+                        mustResolve = false,
+                        operandTypes = signature
+                    )
+                val resolution = resolveCall(cc, ConversionMap())
+                val results = ArrayList<FunctionDef?>()
+                if (resolution != null) {
+                    results.add(resolution.operator.functionDef)
+                }
+                results
+            }
+        }
+    }
+
+    fun resolveCall(callContext: CallContext, conversionMap: ConversionMap): OperatorResolution? {
+        val resolution = operatorMap.resolveOperator(callContext, conversionMap)
+
+        if (resolution != null) {
```

Comment:
Question:  Do we not intend to check for a null resolution.getOperation() like we did in Java?

Reply:
`getOperation` is no longer nullable:
https://github.com/cqframework/clinical_quality_language/blob/master/Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/OperatorResolution.kt#L5

One of the nice thing about the explicit nulls is that the cleanliness is viral. :)